react/features/base/tracks/middleware.js

```
@@ -326,6 +332,29 @@ function _handleNoDataFromSourceErrors(store, action) {
     }
 }
 
+/**
+ * Creates a fake participant for screen share using the track's source name as the participant id.
+ *
+ * @param {Store} store - The redux store in which the specified action is dispatched.
+ * @param {Action} action - The redux action dispatched in the specified store.
+ * @private
+ * @returns {void}
+ */
+function createFakeScreenShareParticipant({ dispatch, getState }, { track }) {
+    const state = getState();
+    const particiipantId = track?.jitsiTrack?.getParticipantId?.();
+    const particiipant = getParticipantById(state, particiipantId);
+    const name = particiipant?.name ? `${particiipant.name}'s screen` : 'Fellow Jitster\'s Screen';
```

log an error and abort if participant is not defined instead of using default